### PR TITLE
BLPOP with a float timeout

### PR DESCRIPTION
**Pull request by michael-grunder**
See #2157

### CODE REVIEW

File: redis_commands.c [hunk 1]
```diff
@@ -1489,8 +1488,9 @@ static int gen_varkey_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
         single_array = argc==1 && Z_TYPE(z_args[0]) == IS_ARRAY;
     } else {
         single_array = argc==2 && Z_TYPE(z_args[0]) == IS_ARRAY &&
-            Z_TYPE(z_args[1]) == IS_LONG;
-        timeout = Z_LVAL(z_args[1]);
+            (Z_TYPE(z_args[1]) == IS_LONG || Z_TYPE(z_args[1]) == IS_DOUBLE);
+        if (single_array)
+            ZVAL_COPY_VALUE(&ztimeout, &z_args[1]);
```
Comment (yatsukhnenko) [inline redis_commands.c:1493]: I suggest to store a pointer to zval here

**Review by yatsukhnenko — CHANGES_REQUESTED** [review]

File: redis_commands.c (hunk 1 above)
Comment (michael-grunder) [reply redis_commands.c:1493]: I made the change locally but it adds complexity to the rest of the function.

Maybe I misunderstand, but If I make `ztimeout` a pointer (defaulted to NULL), then I again have to enclose appending the timeout inside of an `if (timeout != NULL)`, and then add an additional assertion that the timeout is either `IS_DOUBLE` or `IS_LONG`.

Comment (michael-grunder) [reply redis_commands.c:1493]: https://gist.github.com/michael-grunder/1b1a5441b08f3990783e21bcc2051710

Comment (yatsukhnenko) [reply redis_commands.c:1493]: I didn't think about requirement of additional checks. Now I see that current implementation is fine 👍

**Review by yatsukhnenko — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
`double` is not a type in PHP, this should be `float`

**Comment by nicolas-grekas** [comment]
>PHP Warning:  "double" will be interpreted as a class name. Did you mean "float"?

**Pull request merged**